### PR TITLE
Fixed new pylint errors

### PR DESCRIPTION
**Pull request by brandonmbare**

### CODE REVIEW

File: csbuild/__init__.py [hunk 1]
```diff
@@ -563,7 +563,7 @@ class Scope(ContextManager):
 		"""
 		def __init__(self, *scopeTypes):
 			for scopeType in scopeTypes:
-				assert scopeType == Csbuild.ScopeDef.Intermediate or scopeType == Csbuild.ScopeDef.Final, "Invalid scope type"
+				assert scopeType in (Csbuild.ScopeDef.Intermediate, Csbuild.ScopeDef.Final), "Invalid scope type"
```
Comment (ShadauxCat) [inline csbuild/__init__.py]: Was this really a pylint error? This seems like it would be less efficient since it has to create the tuple and iterate through it and then destroy it...

File: csbuild/_build/project_plan.py [hunk 2]
```diff
@@ -337,10 +337,8 @@ def ExecutePlan(self, toolchainName, architectureName, targetName):
 				"all"
 			)
 
-			for depend in flattenedDepends:
-				# pylint: disable=protected-access
-				dependObj = allPlans[depend]
-				# type: ProjectPlan
+			for depend in flattenedDepends: # pylint: disable=protected-access
```
Comment (ShadauxCat) [inline csbuild/_build/project_plan.py]: This is a weird change to me. The disable was supposed to affect this entire for block since there's multiple protected-access instances here. Moving it to the end of the line makes it affect only this line, which isn't accessing anything protected. Why did you move it?

File: csbuild/_testing/functional_test.py [hunk 3]
```diff
@@ -222,7 +222,7 @@ def RunMake(self, *args):
 		log.SetCallbackQueue(callbackQueue)
 
 		class _shared(object):
-			ret = None
+			ret = None, None, None # type: tuple[int, str, str]
```
Comment (ShadauxCat) [inline csbuild/_testing/functional_test.py]: This is also a weird change. Why did you make this a tuple of Nones? If it's related to the type comment you added (not sure why you added that), I thought any type could be made None, including tuple types, but if not, you can do

```
from typing import Optional, Tuple
# type: Optional[Tuple[int, str, str]]
```

File: csbuild/__init__.py (hunk 1 above)
Comment (brandonmbare) [reply csbuild/__init__.py]: Yes, it was giving an error about it.  My bet is they're going for cleaner code?  But, I have no way of knowing for certain.  All I know is it was flagging a new pylint error and I agree the new way does look much better.

Comment (ShadauxCat) [reply csbuild/__init__.py]: Sure, the new way looks better, but where are we drawing the line on cleanliness vs efficiency?

File: csbuild/_build/project_plan.py (hunk 2 above)
Comment (brandonmbare) [reply csbuild/_build/project_plan.py]: The old disabled error still does the same thing.  Appending it to the line still has the effect of applying to the sub-block of that line, so nothing has changed in that regard.

File: csbuild/_testing/functional_test.py (hunk 3 above)
Comment (brandonmbare) [reply csbuild/_testing/functional_test.py]: Pylint has gotten pretty good at determine object usage and in this case, it was flagging the usage of `_shared.ret` not being unpackable.  The only way to fix it was to set the initial value as a tuple of None.

Comment (ShadauxCat) [reply csbuild/_testing/functional_test.py]: I really, really don't like that. Checking if things are None is something we should be able to do. If pylint can't handle initializing to None with a type comment, I would rather disable the pylint check on that than end up in a situation where we're never allowed to use None as a sentinel value.

File: csbuild/__init__.py (hunk 1 above)
Comment (brandonmbare) [reply csbuild/__init__.py]: In this case, the performance impact will be minimal and likely negligible, so I'm opting for readability.  However, I will disable the pylint check on this in the pylintrc since there are valid cases for doing it both ways.
